src/parser.h

```diff
@@ -231,6 +231,9 @@ struct library_data_t {
         /// Used to get the full text of the current job for `status current-commandline`.
         wcstring commandline;
     } status_vars;
+
+    void set_exit_current_script(bool val);
+    void set_returning(bool val);
```

Comment:
Could not figure out how to update these directly from rust, so exposed setters for these

Reply:
https://github.com/fish-shell/fish-shell/pull/9563/commits/a7aa497ff95db61aa8b884371b737b9a02066a2c might be a solution here, but I don't know if it's a good one.

Reply:
Yeah, that also will work, I will leave it like this for now, After your changes are merged then I can update here